src/timedata.cpp

```diff
@@ -40,16 +40,20 @@ static int64_t abs64(int64_t n)
     return (n >= 0 ? n : -n);
 }
 
+#define BITCOIN_TIMEDATA_MAX_SAMPLES 200
+
 void AddTimeData(const CNetAddr& ip, int64_t nOffsetSample)
 {
     LOCK(cs_nTimeOffset);
     // Ignore duplicates
     static set<CNetAddr> setKnown;
+    if (setKnown.size() == BITCOIN_TIMEDATA_MAX_SAMPLES)
+        return;
```

Comment:
@theuni Can you please translate "the early return depends on the quirk" for me? I'm not native English speaker and this is not clear to me. How it should be implemented?